### PR TITLE
Python wrapper with basic unit test

### PR DESCRIPTION
**Pull request by KristerJazz**

### CODE REVIEW

**Comment by KristerJazz** [comment]
@lleon95 I am not sure how to add this to `make check`.
I tried but I couldn't make it work.

Let me know how and I'll push it right away. Thanks!

File: ljmd.py [hunk 1]
```diff
@@ -1,5 +1,164 @@
...
+	def propagate2(self):
+		for i in range(self.natoms):
+			self._ljmd.propagate_velocity(byref(self.sys), i)
+
+	def ekin(self):
+		print("Kinetic energy initialized")
+		self._ljmd.ekin(byref(self.sys))
+	
+	def write_output(self):
+		self.erg.write("% 8d % 20.8f % 20.8f % 20.8f % 20.8f\n" % (self.sys.nfi, self.sys.temp,
```
Comment (lleon95) [inline ljmd.py]: Is there a way to fix this weird long lines?

File: ljmd.py [hunk 2]
```diff
@@ -1,5 +1,164 @@
...
+		self.box = float(data[5])
+		self.restfile = data[6]
+		self.trajfile = data[7]
+		self.ergfile = data[8]
+		self.nsteps = int(data[9])
+		self.dt = float(data[10])
+		self.nprint = int(data[11])
+
+		self.sys = MDSYS_T(natoms=self.natoms,
+							mass=self.mass,
```
Comment (lleon95) [inline ljmd.py]: Could it be the indentation be fixed in a more proper way?

File: ljmd.py [hunk 3]
```diff
@@ -1,5 +1,164 @@
...
+
+	def go(self):
+		for i in range(self.natoms):
+			print(self.sys.rx[i])
 
-so_file = "Obj-serial/ljmd.o"
 
-my_functions = CDLL(so_file)
+if __name__ == '__main__':
+	input_path = "examples/"+sys.argv[1]
```
Comment (lleon95) [inline ljmd.py]: Put a guard here. What happens when you have only one arg?

File: ljmd.py [hunk 4]
```diff
@@ -1,5 +1,164 @@
...
+	output_path = "examples/"
+	print(input_path)
+	so_path = "lib/libljmd.so"
+	#input_path = "examples/argon_108.inp"
```
Comment (lleon95) [inline ljmd.py]: Remove unneeded comments

File: ljmd.py [hunk 5]
```diff
@@ -1,5 +1,164 @@
...
+	main = LJMD(so_path)
+	main.initialize_system(input_path)
+	main.run_simulation()
+	#main.go()
```
Comment (lleon95) [inline ljmd.py]: Remove unneeded comments

File: tests/python_tests/io_test.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import unittest
+from ljmd import *
+
+class IOtest(unittest.TestCase):
+	def test_success(self):
+		assert(5==5)
+
+	def test_fail(self):
+		assert(5==4)
+	#def setUp(self):
```
Comment (lleon95) [inline tests/python_tests/io_test.py]: Remove unneeded comments

File: tests/python_tests/io_test.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import unittest
+from ljmd import *
+
+class IOtest(unittest.TestCase):
+	def test_success(self):
+		assert(5==5)
```
Comment (lleon95) [inline tests/python_tests/io_test.py]: Why this lines?

**Review by lleon95 — CHANGES_REQUESTED** [review]

**Comment by lleon95** [comment]
@KristerJazz 
Could you give me the way to execute the Python Wrapper? I can manage to give you the Makefile lines in this way

File: ljmd.py (hunk 1 above)
Comment (KristerJazz) [reply ljmd.py]: @lleon95 What do you mean?
Do you mean to reformat it with a new line at certain number of columns?

File: ljmd.py (hunk 2 above)
Comment (KristerJazz) [reply ljmd.py]: I am not sure, it doesnt look like that on my screen.

image

File: ljmd.py (hunk 1 above)
Comment (lleon95) [reply ljmd.py]: Exactly. In general, we set the number of columns to 80 if possible

File: tests/python_tests/io_test.py (hunk 7 above)
Comment (KristerJazz) [reply tests/python_tests/io_test.py]: @lleon95 This is outdated. Please check the latest version

**Comment by KristerJazz** [comment]
@lleon95 
The command is,
`python3 ljmd.py examples/argon_108.inp` 

Also you are commenting on outdated commits.

File: ljmd.py [hunk 8]
```diff
@@ -1,5 +1,169 @@
...
+			raise
+
+		assert(len(data)==12), "Invalid input file: File should have 12 rows of input values"
+
+		return data
+	
+	def restart(self, restfile):
+		with open(restfile) as f:
+			r = f.readlines()
+			assert(len(r)==2*self.natoms), "Invalid restart file: File should have rows equal to twice the number of atoms\n"
```
Comment (lleon95) [inline ljmd.py]: is this line OK?

**Review by lleon95 — CHANGES_REQUESTED** [review]

File: ljmd.py (hunk 8 above)
Comment (KristerJazz) [reply ljmd.py]: Yes except for the format. I'll look on how to automate the formatting.

Comment (lleon95) [reply ljmd.py]: Got it

**Comment by lleon95** [comment]
The Makefile will look like (examples/Makefile):

```
# -*- Makefile -*-
include ../Rules.mk

LD_LIBRARY_PATH=$(LIB_PATH)
export LD_LIBRARY_PATH

PERF_SAMPLE=argon_108.inp
PYTHON_EXE=../ljmd.py

default: check

clean:
        rm -f *.xyz *.dat $(PERF_PATH)/*.data $(PERF_RESULT) $(PERF_PATH)/perfstat.txt

check: check_exe check_python

check_python: $(PYTHON_EXE) $(wildcard ../tests/python_tests/*.py)
        python3 $(PYTHON_EXE) argon_108.inp
        python3 $(PYTHON_EXE) argon_2916.inp

check_exe: $(EXE)
        $(EXE) < argon_108.inp
        head -10 argon_108.dat | awk '{printf("%d %.6f %.6f %.6f\n",$$1,$$2,$$3,$$4);}'> a.dat
        head -10 ../reference/argon_108.dat | awk '{printf("%d %.6f %.6f %.6f\n",$$1,$$2,$$3,$$4);}'> b.dat
        cmp a.dat b.dat || exit 1
        $(EXE) < argon_2916.inp
        head -10 argon_2916.dat | awk '{printf("%d %.6f %.6f %.6f\n",$$1,$$2,$$3,$$4);}'> a.dat
        head -10 ../reference/argon_2916.dat | awk '{printf("%d %.6f %.6f %.6f\n",$$1,$$2,$$3,$$4);}'> b.dat
        cmp a.dat b.dat || exit 1
        rm -f a.dat b.dat

perf: $(EXE) $(PERF_RESULT)

$(PERF): $(EXE)
        perf record -o $(PERF) -T -B $(EXE) < $(PERF_SAMPLE)
        perf stat $(EXE) < $(PERF_SAMPLE) -o $(PERF_PATH)/perfstat.txt

$(PERF_RESULT): $(PERF)
        perf report -i $(PERF)
```

File: ljmd.py (hunk 2 above)
Comment (lleon95) [reply ljmd.py]: There should be a thing with the editors... I will check it later

**Comment by lleon95** [comment]
@KristerJazz 
A warning: please, rebase your branch to develop

**Comment by KristerJazz** [comment]
@lleon95 rebasing done.

Can i also add the comparison lines to 
`head -10 argon_108.dat | awk '{printf("%d %.6f %.6f %.6f\n",$$1,$$2,$$3,$$4);}'> a.dat`

Axel told me that the result at the end is somehow chaotic. But the first lines should match. so the comparison lines should work.

**Comment by lleon95** [comment]
Yes, of course. It should be similar to the `check_exe` rule then.

**Review by stefanocampanella — APPROVED** [review]

**Review by lleon95 — APPROVED** [review]

**Pull request merged**